src/HealthGPS/static_linear_model.cpp

```diff
@@ -83,58 +101,133 @@ void StaticLinearModel::update_risk_factors(RuntimeContext &context) {
+        double residual_old = person.risk_factors.at(residual_name);
+        double residual = residuals[i] * info_speed_;
+        residual += sqrt(1.0 - info_speed_ * info_speed_) * residual_old;
+        person.risk_factors.at(residual_name) = residual;
 
         // Update risk factor.
-        double factor_new = linear_factors[i] + residual_new * stddev_[i];
         double expected = get_risk_factor_expected().at(person.gender, names_[i]).at(person.age);
-        factor_new = expected * inverse_box_cox(factor_new, lambda_[i]);
-        person.risk_factors[names_[i]] = factor_new;
+        double factor = linear[i] + residual * stddev_[i];
+        factor = expected * inverse_box_cox(factor, lambda_[i]);
+        person.risk_factors.at(names_[i]) = factor;
+    }
+}
+
+void StaticLinearModel::initialise_policies(Person &person, Random &random,
+                                            ScenarioType scenario) const {
+
+    // Approximate intervention policy values with linear models.
+    auto linear = compute_linear_models(person);
+
+    // Correlated residual sampling.
+    auto residuals = compute_residuals(random, policy_cholesky_);
+
+    // Initialise residuals and intervention policies (do not exist yet).
+    for (size_t i = 0; i < names_.size(); i++) {
+
+        // Initialise residual.
+        auto residual_name = core::Identifier{names_[i].to_string() + "_policy_residual"};
+        double residual = residuals[i];
+        person.risk_factors[residual_name] = residual;
+
+        // Compute intervention policy.
+        double policy = linear[i] + residual;
+        policy = policy_ranges_[i].clamp(policy);
+        double factor_old = person.risk_factors.at(names_[i]);
+        double factor = factor_old * (1.0 + policy / 100.0);
+
+        // Apply intervention policy for intervention scenario.
+        if (scenario == ScenarioType::intervention) {
+            person.risk_factors.at(names_[i]) = factor;
+        }
+    }
+}
+
+void StaticLinearModel::update_policies(Person &person, ScenarioType scenario) const {
+
+    // Approximate intervention policy values with linear models.
+    auto linear = compute_linear_models(person);
+
+    // Initialise residuals and intervention policies (do not exist yet).
+    for (size_t i = 0; i < names_.size(); i++) {
+
+        // Initialise residual.
+        auto residual_name = core::Identifier{names_[i].to_string() + "_policy_residual"};
+        double residual = person.risk_factors.at(residual_name);
+
+        // Compute intervention policy.
+        double policy = linear[i] + residual;
+        policy = policy_ranges_[i].clamp(policy);
+        double factor_old = person.risk_factors.at(names_[i]);
+        double factor = factor_old * (1.0 + policy / 100.0);
+
+        // Apply intervention policy for intervention scenario.
+        if (scenario == ScenarioType::intervention) {
+            person.risk_factors.at(names_[i]) = factor;
+        }
     }
```

Comment:
As far as I can tell, this whole block (actually, the whole of `update_policies` function) only makes sense in the intervention scenario, so I would check if that is the case outside of the function, in the calling code, or just at the beginning of the function, returning immediately if we are not in an intervention scenario without doing anything.

Reply:
As above. I'll see what I can cut out.